dex/farm-staking/src/lib.rs

```diff
@@ -182,6 +185,57 @@ pub trait Farm:
         #[payment_nonce] token_nonce: Nonce,
         #[payment_amount] amount: BigUint,
         #[var_args] opt_accept_funds_func: OptionalArg<ManagedBuffer>,
+    ) -> SCResult<ExitFarmResultType<Self::Api>> {
+        self.unstake_farm_common(
+            payment_token_id,
+            token_nonce,
+            amount,
+            opt_accept_funds_func,
+            false,
+        )
+    }
+
+    #[payable("*")]
+    #[endpoint(unstakeFarmThroughProxy)]
+    fn unstake_farm_through_proxy(
+        &self,
+        #[payment_multi] payments: ManagedVec<EsdtTokenPayment<Self::Api>>,
+    ) -> SCResult<ExitFarmResultType<Self::Api>> {
+        let caller = self.blockchain().get_caller();
+        self.require_whitelisted(&caller);
+
+        require!(payments.len() == 2, "Invalid payments amount");
+
+        let first_payment = payments.get(0).unwrap();
```

Comment:
why do you need 2 payments here ? only one should be enough. first_payment is not used at all.

Reply:
The first payment is the RIDE tokens, which were taken out of liquidity pool by the proxy. They need to be transferred into the staking contract so the user can unbond their funds later.

first_payment.amount should be used somewhere though, you're right. Will fix.

Reply:
`first_payment.amount` is now used at unstake through proxy to create the proper amount of unbond tokens.